### PR TITLE
py-scipy: -fvisibility issue is resolved in 2023.1.0:

### PR DESCRIPTION
**Pull request by eugeneswalker**
CC @rscohn2

### CODE REVIEW

**Review by rgommers — APPROVED** [review]
Ah, great to see that this is fixed in OneAPI. Thanks @eugeneswalker. LGTM.

**Review by kwryankrattiger — DISMISSED** [review]

**Comment by eugeneswalker** [comment]
@spackbot run pipeline

File: var/spack/repos/builtin/packages/py-scipy/package.py [hunk 1]
```diff
@@ -139,7 +139,7 @@ class PyScipy(PythonPackage):
     # https://github.com/mesonbuild/meson/pull/10909#issuecomment-1282241479
     # Intel OneAPI ifx claims to support -fvisibility, but this does not work.
     # Meson adds this flag for all Python extensions which include Fortran code.
-    conflicts("%oneapi", when="@1.9:")
+    conflicts("%oneapi@:2023.0.0", when="@1.9:")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-scipy/package.py]: Resolved in 2023.1.0 or in 2023.0.1? Might want:
```suggestion
    conflicts("%oneapi@:2023.0", when="@1.9:")
```

Comment (eugeneswalker) [reply var/spack/repos/builtin/packages/py-scipy/package.py]: Yes, thank you!

**Review by adamjstewart — APPROVED** [review]

**Comment by eugeneswalker** [comment]
@scottwittenburg This is an example of llvm being broken on `ml-linux-x86_64-rocm`

This PR changes only py-scipy, and py-scipy is not a dependency of llvm. So the llvm build failure in CI seems unrelated to the changes proposed in this PR.

How should we proceed here?

**Comment by scottwittenburg** [comment]
The current pruning policy for specs in PR pipelines is to traverse up 1 level from changed packages (to direct dependents) and then remove from pruning candidacy any specs in the resulting DAG(s).  Since `py-keras` is a direct dependent of `py-scipy` and `llvm` is a transitive dependency of `py-keras`, we didn't prune `llvm` from this pipeline.

However, it seems the most recently gitlab-tested merge commit for this PR doesn't include a fix from last week (#39566).  If you rebase on `deeeb860`, your pipelines should run right away and I think the `llvm` issues will go away.

**Pull request merged**